### PR TITLE
Closes #868: Share images with Open Food Facts

### LINKED ISSUE #868: Let people share images with barcodes to Open Food Facts
Sometimes people don't have the time to open the app and properly scan a product. Instead, they take  2 or 3 photos of the product, including one with the barcode. Let the people add those pictures to Open Food Facts, and then open a wizard to help them add those pictures to an existing product or create  a new product, based on the case.

If the user only selects one image, it should ensure he has no more additional pictures he can share

image

### PR DESCRIPTION
**Pull request by sowmya9029**
## Description
Screen shots are mentioned in #868 
Will add unit tests to the pr.
Localization for the added strings is not present, looking for inputs on how to add them.

Describe the changes made and why they were made instead of how they were made.
Did the changes and workflow as mentioned in #868
## Related issues and discussion
#868 
 

 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [ ] Code is well documented
 - [ ] Include unit tests for new functionality
 - [ ] All user-visible strings are made translatable
 - [ ] Code passes Travis builds in your branch
 - [ ] If you have multiple commits please combine them into one commit by squashing them.
 - [ ] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by teolemon** [comment]
I'm changing the name of the PR to be more explicit.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -479,4 +479,12 @@
     <string name="label_string">Label</string>
     <string name="category_string">Category</string>
 
+    <!-- strings related to share images -->
+    <string name="sorry_msg">Sorry we cannot add pictures without barcode from mobile.Please try web</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Missing space: "from mobile. Please try web"

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -479,4 +479,12 @@
...
+    <string name="no_barcode">We have not detected any barcode</string>
+    <string name="enter_barcode">Please type barcode below</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Please type the barcode below

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -479,4 +479,12 @@
...
+    <string name="format_error">Wrong Barcode/QR format</string>
+    <string name="code_detected">We have detected the bar code</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: We have detected the barcode

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -479,4 +479,12 @@
...
+    <string name="do_you_want_to">Do you want to add photos to this bar code?</string>
+
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Do you want to add photos to this barcode?

**Comment by teolemon** [comment]
@sowmya9029 Tiny conflict. Can you fix it ?
Also, can you tweak the strings as requested ?
Also, can you share screenshots with the latest tweaks here (especially when no barcode is detected)

**Comment by sowmya9029** [comment]
Here are the screenshots
device-2018-03-14-131158
device-2018-03-14-131240

**Comment by Karljoones** [comment]
Maybe for the dialog were the barcode is found it should be the barcode as the title and the message should be something like `A barcode was detected. Do you want to add photos?`

**Comment by teolemon** [comment]
@sowmya9029 @Karljoones: 
for unrecognized picture, we should display it:
image

**Comment by sowmya9029** [comment]
So we should even show the selected image in the alert box if there is no barcode. Am I correct?

**Comment by Karljoones** [comment]
@sowmya9029 @teolemon I agree, we should display the image in the dialog even if there is no barcode detected. Since the system may not detect the barcode 100% of the time, the user can read the barcode off the image if it is there but the system has not detected it. There should be a bit more padding around the edit text too

**Comment by sowmya9029** [comment]
@Karljoones  I agree. Working on the change.

**Comment by teolemon** [comment]
Merge conflict to solve.
This one would be nice to merge asap, since it's very useful.

**Comment by Karljoones** [comment]
Conflicts resolved

**Comment by Karljoones** [comment]
Merged, thanks @sowmya9029

**Comment by sowmya9029** [comment]
Thanks @Karljoones for resolving this. I was on vacation and dint see this comment.

**Pull request merged**